### PR TITLE
Users/v muuppugundu/zoomreportsfixes

### PR DESCRIPTION
**Pull request by v-muuppugund**
Required items, please complete
   
   Change(s):
   - Zoom reports Authentication changes

   Reason for Change(s):
   - There is a change in Zoom authentication

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - Yes,Please find below screen shots for reference
   - 
image
image
image

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py [hunk 1]
```diff
@@ -11,95 +10,186 @@
...
             logging.info("The last time point is: {}".format(past_time))
         else:
-            logging.info("There is no last time point, trying to get events for last week.")
-            past_time = (current_time_day - datetime.timedelta(days=7)).strftime("%Y-%m-%d")
+            logging.info(
+                "There is no last time point, trying to get events for last week.")
+            past_time = (current_time_day -
+                         datetime.timedelta(days=7)).strftime("%Y-%m-%d")
         state.post(current_time_day.strftime("%Y-%m-%d"))
         return (past_time, current_time_day.strftime("%Y-%m-%d"))
```
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:121]: We should able to process back dated data with defined window.

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:1]: We should stop processing the records if we reached near time out of function.

File: Solutions/ZoomReports/Data Connectors/ZoomAPISentinelConn.zip
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomAPISentinelConn.zip:1]: We should create V2 version not to impact existing customers.

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:1]: Added the logic and working on V2.0 zip file and short link, will be testing it, once done resolve it.

File: Solutions/ZoomReports/Data Connectors/ZoomAPISentinelConn.zip
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomAPISentinelConn.zip:1]: Working on it

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py [hunk 2]
```diff
@@ -148,35 +271,73 @@ def post_data(self, body, chunk_count):
...
         element['event_name'] = api_req_name
         results_array.append(element)
 
-def get_main_info():
+def check_if_functiontime_is_over(start_time, interval_minutes, max_script_exec_time_minutes):
+    """Returns True if function's execution time is less than interval between function executions and
+    less than max azure func lifetime. In other case returns False."""
+    
+    logging.info("started Max function time check")
+    max_minutes = min(interval_minutes, max_script_exec_time_minutes)
```
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:297]: Update it to min_minutes

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py [hunk 3]
```diff
@@ -148,35 +271,73 @@ def post_data(self, body, chunk_count):
...
+    if max_minutes > 1:
```
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:298]: Can you please re-check this logic.

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py [hunk 4]
```diff
@@ -9,97 +8,197 @@
...
 customer_id = os.environ['WorkspaceID']
 shared_key = os.environ['WorkspaceKey']
 connection_string = os.environ['AzureWebJobsStorage']
 logAnalyticsUri = os.environ.get('logAnalyticsUri')
 table_name = "Zoom"
 chunksize = 10000
+retry = 3 ## To do : need to move function app configuration
+error=False
+#Max script execution
+SCRIPT_EXECUTION_INTERVAL_MINUTES = 30
```
Comment (daxesht) [inline Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:26]: Moving this to function app configuration.

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py (hunk 2 above)
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:297]: Done the changes, working on local testing with mock end point

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py (hunk 3 above)
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:298]: Rechecked the logic and modified else condition and working on local testing with mock end point

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py (hunk 4 above)
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:26]: Done ,moved to function app config,working on local testing,once local testing completed with mock end point,will deploy and test the new and old version

Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py]: Deployed and tested,also tested locally with mock end point

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py (hunk 3 above)
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py]: Locally tested with  mock end point

**Review by daxesht — DISMISSED** [review]

File: Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py [hunk 5]
```diff
@@ -11,95 +10,179 @@
...
-            past_time = (current_time_day - datetime.timedelta(days=7)).strftime("%Y-%m-%d")
+            logging.info(
+                "There is no last time point, trying to get events for last week.")
+            past_time = (current_time_day -
+                         datetime.timedelta(days=7)).strftime("%Y-%m-%d")
         state.post(current_time_day.strftime("%Y-%m-%d"))
         return (past_time, current_time_day.strftime("%Y-%m-%d"))
 
-    def get_report(self, report_type_suffix,next_page_token = None):
+    def get_report(self, report_type_suffix, next_page_token=None):
```
[the remark this answers is not in the document]
Comment (v-muuppugund) [reply Solutions/ZoomReports/Data Connectors/ZoomSentinelConnector/__init__.py:175]: /package

**Review by daxesht — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**